src/core/dbus-execute.c

```diff
@@ -1775,20 +2146,43 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
-        } else if (streq(name, "SELinuxContext")) {
+        } else if (STR_IN_SET(name, "SELinuxContext", "AppArmorProfile", "SmackProcessLabel")) {
+                bool ignore;
                 const char *s;
 
-                r = sd_bus_message_read(message, "s", &s);
+                r = sd_bus_message_enter_container(message, 'r', "bs");
```

Comment:
@yuwata uh, yeah it should... and we fucked that up when we first added support for this. It's my fault... But at this point it's API I fear, and we shouldn't change that I guess.